### ISSUE TITLE
BCES00081 (Killzone 2) A freeze at the cutscene right after killing Radec

### ISSUE THREAD

**Issue by Emogop** [RPCS3/rpcs3#13209]
### Quick summary

It's right at the end of the game which makes it even more disappointing
Because the game as a whole runs pretty good with the patch enabled and with good performance.
One interesting thing is when the freeze occurs the CPU usage go up from its usual 50-60% to a solid 100% all cores and CPU fan spinning louder.

### Attach a log file

[RPCS3 killzone2.zip](https://github.com/RPCS3/rpcs3/files/10389129/RPCS3.killzone2.zip)

### Attach capture files for visual issues

Here is the save files to recreate the issue.
[BCES00081-KILLZONE2.zip](https://github.com/RPCS3/rpcs3/files/10389139/BCES00081-KILLZONE2.zip)

But be aware! First you need to kill the son of a bitch and he's hard as hell on a normal difficulty.

### System configuration

CPU: i9 9980K ES 4.3 Ghz 8/8 HT OFF
RAM: 16 Gb
GTX 1060 
win 10 22H2

**Comment by Asinin3** [comment]
Does it happen without patches, and is it reproduceable every time?

**Comment by Emogop** [comment]
I haven't tried it without a disable SPU post processing patch to be honest. But I managed to reproduce it 2 times with the same exact result.
Also I searched the discord and found a guy with the exact same problem. Only he managed to beat the boss 3 times and still get the freeze.

**Comment by SoluS-RUS** [comment]
I can also reproduce this issue every time. I have the save state just before this moment. I do not have any patches enabled.
Please see log attached.
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/10499462/RPCS3.log.gz)

**Comment by illusion0001** [comment]
Try interpreters

**Comment by SoluS-RUS** [comment]
Tried "Static interpreter" option both for PPU and SPU. The same freeze at exactly the same moment (but at 2 FPS now :) )

**Comment by deatrocker** [comment]
The same issue, and everytime you must kill radec again and again =\ there is no checkpoint before boss fight, nice work guerilla, not even thinking about unstable emulator's gamers =)

**Comment by Aquakill1988** [comment]
Happens to me too I have the BCUS98166 version of the game, do you think there will be any updates on this game soon?

**Comment by PawelGribov** [comment]
Also reproduced on BCES00081, a freeze happens during cutscene right after Rico's "Visari!" on any configurations/patches.

**Comment by CoffeeXpress1** [comment]
Sorry, but where i put the files that you send?

**Comment by digant73** [comment]
could someone prove the game save for BCES0081 so I will try to reproduce it?

**Comment by milkowski** [comment]
I can confirm the issue. 100% reproducible with PPU Decorer set as "LLVM Recompiler", does not occur when set as "Interpreter (static)". I'm attaching the last game save, just before killing Radec.
[BCES00081-KILLZONE2.zip](https://github.com/user-attachments/files/17090902/BCES00081-KILLZONE2.zip)

**Comment by alexconsole04** [comment]
> I can confirm the issue. 100% reproducible with PPU Decorer set as "LLVM Recompiler", does not occur when set as "Interpreter (static)". I'm attaching the last game save, just before killing Radec. [BCES00081-KILLZONE2.zip](https://github.com/user-attachments/files/17090902/BCES00081-KILLZONE2.zip)

Yep, that's the fix to this issue. Just tried it on BCUS98116 and the game did not freeze. LLVM Recompiler is the issue it seems. This is without patches enabled and just the wiki settings used.
[RPCS3.log.gz](https://github.com/user-attachments/files/18250468/RPCS3.log.gz)

![Screenshot 2024-12-26 003005](https://github.com/user-attachments/assets/c34ebcfc-4fbd-445b-b3ce-952ff89cb2a0)

**Comment by JonathanMaillard** [comment]
Just played the game for the fist time on RPCS3 with the wiki settings and LLVM Recompiler, the cutscene (and the whole game) worked flawlessly.

Maybe it should be retested to see of if it still happens on other PCs. I played the BCES disk with all the patches.

9800X3D/RTX4080

**Comment by alexconsole04** [comment]
It still freezes, regardless of patches, the issue has not been fixed

[RPCS3.log.gz](https://github.com/user-attachments/files/19219708/RPCS3.log.gz)

**Comment by alexconsole04** [comment]
This is still an issue on latest. Tested with clean PPU cache.

<img width="2560" height="1392" alt="Image" src="https://github.com/user-attachments/assets/6b8de7d2-4598-4977-8859-7bbd861191b9" />

[RPCS3.zip](https://github.com/user-attachments/files/23841456/RPCS3.zip)

**Comment by thiagom123** [comment]
> This is still an issue on latest. Tested with clean PPU cache.
> 
> <img alt="Image" width="2000" height="1392" src="https://private-user-images.githubusercontent.com/145717541/520518513-6b8de7d2-4598-4977-8859-7bbd861191b9.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjY5NTkyNDksIm5iZiI6MTc2Njk1ODk0OSwicGF0aCI6Ii8xNDU3MTc1NDEvNTIwNTE4NTEzLTZiOGRlN2QyLTQ1OTgtNDk3Ny04ODU5LTdiYmQ4NjExOTFiOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIyOFQyMTU1NDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kNDM5YTc0NmUyYzliNTQ1NTNlOGE1YzVhN2U0YTNiMDIxY2U2YTY4MjQyNjVjYzYyNjI2YWFiYmUzM2Q4NGFlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.-i2mg7eW4B-MfHFedX1Afkt81fBLi7rpHp9UZ0JXiUM">
> [RPCS3.zip](https://github.com/user-attachments/files/23841456/RPCS3.zip)

Had the same issue today on current build (0.0.38-18580), I think it crashed pretty much at the same moment/frame.

**Comment by alexconsole04** [comment]
This is still an issue. 

<img width="2560" height="1392" alt="Image" src="https://github.com/user-attachments/assets/d77acfa7-d2ad-4f57-b88c-9b662e894a62" />

[RPCS3.zip](https://github.com/user-attachments/files/29654191/RPCS3.zip)
